js-api/src/chem.ts

```diff
@@ -261,12 +264,17 @@ export namespace chem {
       const extractor = extractors
         .find((f) => new RegExp(f.options['inputRegexp']).test(x));
 
-      if (extractor != null && !checkSmiles(x) && !isMolBlock(x))
-        extractor
-          .apply([new RegExp(extractor.options['inputRegexp']).exec(x)![1]])
-          .then((mol) => this.setMolecule(mol));
-      else
-        this.setMolecule(x);
+        if (extractor != null && !checkSmiles(x) && !isMolBlock(x)) {
+          this.calculating = true;
+          extractor
+            .apply([new RegExp(extractor.options['inputRegexp']).exec(x)![1]])
+            .then((mol) => {
```

Comment:
You forgot about handling the exception, which is quite likely in this case

Reply:
I tried to wrap in try{}catch{}. But in case we are using extractors from some package (which is always the case) and this extractor throwed an exception, this exception has never been catched by this catch in chem.ts. Anyway, I will put try catch block back.

Reply:
> But in case we are using extractors from some package (which is always the case) and this extractor throwed an exception, this exception has never been catched by this catch in chem.ts.

Not sure what you meant? We would not catch it here? Why?